cmd/google_cloud_ops_agent_engine/main.go

```diff
@@ -31,6 +32,36 @@ var (
 	stateDir = flag.String("state", "/var/lib/google-cloud-ops-agent", "path to store agent state like buffers")
 )
 
+func runStartupChecks(service string) error {
```

Comment:
Yes, they are doing the same thing. The only thing that differs is service `logger` objects that differ in the Windows and Linux for the main service. I could just put the logging logic outside `runStartupChecks`. What do you suggest ?

Reply:
The two functions are a bit more different/shorter now so it's probably not necessary anymore. One thing you could do to avoid duplication of the registry at least is to move that into a factory function, e.g.

```go
function healthCheckRegistryFactory() healthchecks.HealthCheckRegistry {
	return healthchecks.HealthCheckRegistry{
		healthchecks.PortsCheck{},
		healthchecks.NetworkCheck{},
		healthchecks.APICheck{},
	}
}
```

Up to you.

Reply:
Created the `HealthCheckRegistryFactory` in https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/a0d452890652355f4c39c80900de606aed109e65 .